libcodechecker/analyze/store_handler.py

```diff
@@ -420,6 +386,23 @@ def addReport(storage_session,
             str(ex))
 
 
+def changePathAndEvents(storage_session, run_id, report_path_map):
+    session = storage_session.get_transaction(run_id)
+    report_ids = report_path_map.keys()
+
+    session.query(BugPathEvent) \
+        .filter(BugPathEvent.report_id.in_(report_ids)) \
+        .delete(synchronize_session=False)
+
+    session.query(BugReportPoint) \
+        .filter(BugReportPoint.report_id.in_(report_ids)) \
+        .delete(synchronize_session=False)
+
+    for report_id, (bug_path, events) in report_path_map.items():
+        store_bug_path(session, bug_path, report_id)
```

Comment:
Just a note, in the future we might want to experiment with bulk insert APIs of sqlalchemy for the bottlenecked inserts.